crates/fluvio-test/src/tests/smoke/mod.rs

```diff
@@ -68,14 +77,90 @@ impl TestOption for SmokeTestOption {
 //}
 
 #[fluvio_test(topic = "test")]
-pub async fn smoke(mut test_driver: Arc<FluvioTestDriver>, mut test_case: TestCase) -> TestResult {
+pub fn smoke(mut test_driver: FluvioTestDriver, mut test_case: TestCase) -> TestResult {
```

Comment:
So I think we either have two procedure macro or one procedure macro with different attribute.  If for example, attribute as in `#[fluvio_test(topic = 'test', async)]` this will inject async (in the election test), otherwise it will generate raw code